client/src/views/files/FoldersPage.vue

```diff
@@ -345,18 +340,21 @@ async function importFiles(): Promise<void> {
 }
 
 function selectAllFiles(checked: boolean): void {
-  for (const item of fileItemRefs.value || []) {
-    item.isSelected = checked;
-    item.showCheckbox = checked;
+  if (displayView.value === DisplayState.List) {
```

Comment:
I meant handle the Grid case by selecting/unselecting all items too.

You may be able to do something like
```typescript
for (const item of displayView.value === DisplayState.List ? fileListItemRefs.value : fileGridItemRefs.value) {
      item.isSelected = checked;
      item.showCheckbox = checked;
}
allFilesSelected.value = checked;
```

Reply:
I understand, but in the grid case we don't have a button to check everything, so why do we need to add a grid case in the condition?

Reply:
> I know, you can't select all files in GridMode, but this function might be used to reset the selection at one time and it should handle GridMode too.

The function is called `selectAllFiles`, not `selectAllFilesInListMode`, and it should handle both cases to avoid bugs further down the line where someone is calling this function to reset the selection for example, and forgets to check the result in grid mode.